javaagent-bootstrap/src/main/java/io/opentelemetry/javaagent/bootstrap/HelperResources.java

```diff
@@ -17,36 +24,70 @@
  */
 public final class HelperResources {
 
-  private static final Cache<ClassLoader, Map<String, URL>> RESOURCES = Cache.weak();
-  private static final Map<String, URL> ALL_CLASSLOADERS_RESOURCES = new ConcurrentHashMap<>();
+  private static final Cache<ClassLoader, Map<String, List<URL>>> RESOURCES = Cache.weak();
+  private static final Map<String, List<URL>> ALL_CLASSLOADERS_RESOURCES =
+      new ConcurrentHashMap<>();
 
   /**
-   * Registers the {@code url} to be available to instrumentation at {@code path}, when given {@code
-   * classLoader} attempts to load that resource.
+   * Registers the {@code urls} to be available to instrumentation at {@code path}, when given
+   * {@code classLoader} attempts to load that resource.
    */
-  public static void register(ClassLoader classLoader, String path, URL url) {
-    RESOURCES.computeIfAbsent(classLoader, unused -> new ConcurrentHashMap<>()).put(path, url);
+  public static void register(ClassLoader classLoader, String path, Enumeration<URL> urls) {
+    RESOURCES
+        .computeIfAbsent(classLoader, unused -> new ConcurrentHashMap<>())
+        .compute(path, (k, v) -> append(v, urls));
+  }
+
+  /** Registers the {@code urls} to be available to instrumentation at {@code path}. */
+  public static void registerForAllClassLoaders(String path, Enumeration<URL> urls) {
```

Comment:
how about accepting `List<URL>` instead of `Enumeration` and calling `Collections.list()` at the caller (just for the sake of keeping `Enumeration` out of public API)?